kfdef/odh-core-experimental.yaml

```diff
@@ -0,0 +1,96 @@
+# The components listed in this KfDef are pulling the latest kustomize manifest and images directly from each components development repo
+# If you deploy odh-core using this KfDef there is no expectation for stability and components made be updated without any prior notice
+apiVersion: kfdef.apps.kubeflow.org/v1
+kind: KfDef
+metadata:
+  name: odh-core-experimental
+spec:
+  applications:
+  - kustomizeConfig:
+      repoRef:
+        name: manifests
+        path: odh-common
+    name: odh-common
+  - kustomizeConfig:
+      overlays:
+      - authentication
+      # Uncomment the odhdashboard overlay below to have the operator deploy the configs
+      # This should be removed from the deployed kfdef immediately after the operator deploys the configs
+      # This will prevent the operator from reconciling the configs when they have been modified externally
+      #- odhdashboardconfig
+      repoRef:
+        name: manifests-dashboard  # Use the odh-dashboard repo as the source for this kustomize manifest
+        path: manifests
+    name: odh-dashboard
+  - kustomizeConfig:
+      repoRef:
+        name: manifests
+        path: prometheus/cluster
+    name: prometheus-cluster
+  - kustomizeConfig:
+      repoRef:
+        name: manifests
+        path: prometheus/operator
+    name: prometheus-operator
+  - kustomizeConfig:
+      repoRef:
+        name: manifests
+        path: grafana/cluster
+    name: grafana-cluster
+  - kustomizeConfig:
+      repoRef:
+        name: manifests
+        path: grafana/grafana
+    name: grafana-instance
+  # Deploy the KF notebook controller and ODH controller separately since they are normally bundled into the
```

Comment:
```
time="2022-09-28T16:27:08Z" level=error msg="Error evaluating kustomization manifest for kf-notebook-controller: accumulating resources: recursed merging from path 'overlays/openshift': may not add resource with an already registered id: rbac.authorization.k8s.io_v1_ClusterRole|~X|notebook-controller-role"
```

Reply:
@cfchase I moved this experimental kfdef work to #689 since there are issues with pulling from multiple repos that shouldn't block this PR